tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/concurrent/ControlledWorkerScheduler.java

```diff
@@ -112,8 +112,10 @@ public int getNumberOfTasks() {
 
 	protected void initWorkerThreads() {
 
-		executor = new ThreadPoolExecutor(Math.min(10, nWorkers / 2), nWorkers, 30L, TimeUnit.SECONDS, _taskQueue,
+		ThreadPoolExecutor _executor = new ThreadPoolExecutor(nWorkers, nWorkers, 60L, TimeUnit.SECONDS, _taskQueue,
 				new NamingThreadFactory(name));
+		_executor.allowCoreThreadTimeOut(true);
+		this.executor = _executor;
```

Comment:
Btw. Unrelated to this particular PR. I noticed a lot of Protected variables in this class, do you need them to be available so broadly.

Also, I was wondering what happens if you call initWorkerThreads multiple times? Does it create a bunch of thread pools?

Reply:
@hmottestad I agree that some of these variables should be made private, but this close to release I'd rather have confirmation from @aschwarte10 that that won't break dependent code. 

As for your second remark: it looks as if the intent is that `initWorkerThreads` is only invoked from the constructor. So that would happen only once.

Reply:
I agree that methods should be private for this scheduler (and also the initialization could be done in a better way). The current implementation structure is mostly for historical reasons (the initial code is almost 8 years old by now) and quite some versions ago the extension hierarchy for the schedulers looked different (e.g. for union and join) making it required to have certain things protected.
As suggested by Jeen we can have another push on this PR to make the methods private